### PR TITLE
add step for quay interop tests on OPP

### PR DESCRIPTION
**Pull request by dongboyan77**
Draft a specific test step for quay interop tests on OPP

### CODE REVIEW

**Comment by dongboyan77** [comment]
@ascerra Hi, actually we already have test step for existing quay, see https://github.com/openshift/release/tree/master/ci-operator/step-registry/quay-tests/test-quay-e2e
There are many configurations needed to cover for quay. That means some cases are not available to OPP. We don't attempt to involve these cases into OPP tests. So I draft a specific step for OPP.

**Review by gparvin — APPROVED** [review]
Needs a lot of CPU and memory but this looks good to me.

**Comment by ascerra** [comment]
> @ascerra Hi, actually we already have test step for existing quay, see https://github.com/openshift/release/tree/master/ci-operator/step-registry/quay-tests/test-quay-e2e There are many configurations needed to cover for quay. That means some cases are not available to OPP. We don't attempt to involve these cases into OPP tests. So I draft a specific step for OPP.

Alright that makes sense, can we maybe just use interop instead of opp. So this test step will be able to be re-used by anyone that wants to make use of some quay interop tests.

File: ci-operator/step-registry/quay-tests/quay-opp-interop/quay-tests-quay-opp-interop-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  commands: quay-tests-quay-opp-interop-commands.sh
+  resources:
+        requests:
+          cpu: "2"
+          memory: 4Gi
+  timeout: 1h0m0s
+  grace_period: 15m0s
+  documentation: |-
+    Execute QUAY Interop Testing for OPP
+  #env:
```
Comment (ascerra) [inline ci-operator/step-registry/quay-tests/quay-opp-interop/quay-tests-quay-opp-interop-ref.yaml:17]: We should remove this env var if its not being used

**Comment by dongboyan77** [comment]
@ascerra Hi, I renamed the step, please review and add /lgtm label. Thanks

**Comment by dongboyan77** [comment]
/pj-rehearse periodic-ci-stolostron-policy-collection-main-opp-ocp4.12-interop-openshift-plus-interop-aws

**Comment by dongboyan77** [comment]
/pj-rehearse periodic-ci-stolostron-policy-collection-main-opp-ocp4.12-interop-openshift-plus-interop-aws

**Comment by ascerra** [comment]
The [XML being generated](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/38385/rehearse-38385-periodic-ci-stolostron-policy-collection-main-opp-ocp4.12-interop-openshift-plus-interop-aws/1648654379428876288/artifacts/openshift-plus-interop-aws/quay-tests-quay-interop-test/artifacts/quay_automation_testiong_report-805efae426bde44ae5f27d7bbcdd2629.xml) is showing that there were 7 test failures.

Do we want to resolve these before merging this?

**Comment by gparvin** [comment]
I see the tests are passing now and this looks ready to me: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/38385/rehearse-38385-periodic-ci-stolostron-policy-collection-main-opp-ocp4.12-interop-openshift-plus-interop-aws/1649073459143118848/artifacts/openshift-plus-interop-aws/quay-tests-quay-interop-test/artifacts/quay_automation_testiong_report-821dea36553323b3191789a44c8478f9.xml
/lgtm

**Pull request merged**